app/service/data_svc.py

```diff
@@ -49,8 +49,14 @@ def __init__(self, dao):
             await self.dao.create('core_parser', parser)
         return 'Saved ability: %s' % id
 
-    async def create_adversary(self, name, description, phases):
-        identifier = await self.dao.create('core_adversary', dict(name=name.lower(), description=description))
+    async def create_adversary(self, name, description, phases, locked):
+        ident = await self.dao.get('core_adversary', dict(name=name.lower()))
+        if len(ident) == 0:
+            identifier = await self.dao.create('core_adversary', dict(name=name.lower(), description=description, locked=locked))
+        else:
+            if ident[0]['locked']:
+                return 'Adversary %s locked... skipping...' % name
+            identifier = ident[0]['id']
         await self.dao.delete('core_adversary_map', dict(adversary_id=identifier))
```

Comment:
this could be moved to its own function 'remap_adversary_abilities...'

Reply:
ooh yeah i like that better. i try to avoid if/statements wherever possible (i like them within comprehensions). trying to limit if/else blocks is a good goal. obviously, they're a useful part of the language but they can also be indicative of code trying to do too much.

Reply:
I've split things up so they're more isolated to specific purpose.